src/client.rs

```diff
@@ -41,20 +43,31 @@ impl Client {
         if !base_url.ends_with('/') {
             base_url += "/"
         }
+        let http = Self::http_client_builder(user_agent)
+            .build()
+            .context("failed to initialize HTTP client")?;
+        let http_no_sni = Self::http_client_builder(user_agent)
+            .tls_sni(false)
+            .build()
+            .context("failed to initialize no-sni HTTP client")?;
+
         Ok(Self {
             account_id,
             base_url,
             cached_auth_token: Arc::new(Mutex::new(None)),
-            http: reqwest::Client::builder()
-                .timeout(Duration::from_secs(60))
-                .read_timeout(Duration::from_secs(10))
-                .user_agent(user_agent)
-                .build()
-                .context("failed to initialize HTTP client")?,
+            http,
+            http_no_sni,
```

Comment:
It is unfortunate that we need two clients so can't share DNS cache and other things. But I guess that is just how it is.